app/containers/WriteReviewPage/GuideReviewCard.js

```diff
@@ -7,52 +7,42 @@ import PageSection from 'components/PageSection';
 import BorderBottomDiv from 'components/shared/BorderBottomDiv';
 import Stars from 'components/Stars';
 
-class GuideReviewForm extends React.Component {
+class GuideReviewCard extends React.Component {
   constructor(props) {
     super(props);
-    this.handleButtonClick = this.handleButtonClick.bind(this);
-    this.state = {
-      isGuided: props.isGuided,
-    };
   }
 
   componentWillMount() {
-    if(this.props.isGuided) {
-      this.props.getGuideList();  // lifting up state: API call is handled by WriteReviewPage container
+    // Grab guide list immediately (API call) if somehow we know it's guided but there's no guide selected.
+    if(this.props.isGuided && (this.props.selectedGuideId !== null)) {
+      this.props.getGuideList();
     }
   }
 
-  shouldComponentUpdate(nextProps, nextState) {
-    // Update if...
-    const doUpdate = ((nextState.isGuided !== this.state.isGuided) //...switching between yes/no options (affects disabled status for Select component)
-      || (nextProps.guideOptions.length !== this.props.guideOptions.length) //...or guide list is updated (affects Select component's options)
-      || (nextProps.selectedGuideId !== this.props.selectedGuideId) //...or a guide is selected (affects Select component and disabled status for Stars component)
-    );
-
-    return doUpdate;
-  }
-
-  componentWillUpdate(nextProps) {
+  componentWillReceiveProps(nextProps) {
     // Do another API call if, somehow, the options array's length changes
-    if(this.props.guideOptions.length && nextProps.guideOptions.length !== this.props.guideOptions.length) {
-      this.props.getGuideList();  // lifting up state: API call is handled by WriteReviewPage container
+    if(this.props.guideOptions.length && (nextProps.guideOptions.length !== this.props.guideOptions.length)) {
```

Comment:
How would this ever happen? I'm confused about this, wouldn't this cause an infinite loop?

Reply:
This was a precaution I thought of in case maybe the data for the component was changed (either by an update to the guide list or some user/coder trying to tweak stuff). This doesn't cause an infinite loop because an update when they are unequal will make them equal:
```
// Let's say Redux state's guide list is length 3, so this.props.gOpts.length = 3:
nextProps.gOpts.length = 4  // Let's say this is passed in
if(nextProps.gOpts.length !== this.props.gOpts.length) {  //true, 4 !== 3
  getGuideList();  //This reinforces putting state list onto props, so this.props.gOpts.length is still 3
}
/* Updating the guideOptions with getGuideList will call componentWillReceiveProps again, but now nextProps.gOpts.length = 3, which is equal to this.props.gOpts.length.

Let's say our state's options did change, so nextProps is actually the true length and different to what's in this.props. This will still update this.props.gOpts.length and call componentWillReceiveProps using the same length it updated with, so nextProps.gOpts.length will equal this.props.gOpts.length again. This not only updates for unexpected changes, but seems to reinforce that what we have in the component props matches what's in Redux state. */
```
Not sure if that was an appropriate use of a code block, but hopefully that's at least a good (and correct) explanation. I'm not sure where this would come up in practice, but it seemed like a good check for "just in case."